src/s3/util/util.ts

```diff
@@ -0,0 +1,18 @@
+/*!
```

Comment:
Would suggest this lives in `src/s3/util.ts`. We should not have lots of service-specific util things, so a subdirectory should never be needed in a service namespace.

Reply:
I think all of those changes are reasonable, done